omc3/run_kmod.py

```diff
@@ -285,8 +285,11 @@ def create_lsa_results_file(betastar_required, instruments_found, results_df, in
         exporting_columns=['LABEL', f'{BETA}{STAR}X', f'{ERR}{BETA}{STAR}X', f'{BETA}{STAR}Y', f'{ERR}{BETA}{STAR}Y']
         lsa_results_df=results_df[exporting_columns].rename(columns=dict(zip(exporting_columns, LSA_COLUMNS)))
     if instruments_found:
-        lsa_results_df = pd.concat([lsa_results_df, instrument_beta_df],
-                                   axis="index", sort=False, ignore_index=True)
+        # We first make sure we don't try a concat operation if a df is empty
+        # (otherwise pandas complains with a FutureWarning since 2.1.1)
+        dfs_to_concat = [lsa_results_df, instrument_beta_df]
+        dfs_to_concat = list(filter(lambda df: not df.empty, dfs_to_concat))
+        lsa_results_df = pd.concat(dfs_to_concat, axis="index", sort=False, ignore_index=True)
```

Comment:
This fixes the `FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.`

I added a simple check where only non-empty dataframes from these two are passes to the concatenation.

Reply:
I think list comprehension in this case is more readable and probably faster (not that this matters).
```
dfs_to_concat = [df for df in dfs_to_concat if not df.empty()]
```

But then also:
```
if not len(dfs_to_concat):
    msg = "All dfs are empty! Something went wrong along the way."
    raise ValueError(msg)
 ```

Optional
```
if len(dfs_to_concat) == 1:
    lsa_results_df = dfs_to_concat[0]
else:
    ...
```